src/main/resources/com/tresys/pcap/xsd/pcap.dfdl.xsd

```diff
@@ -397,29 +395,23 @@ The message root is 'PCAP'.
           else if (fn:exists(../EchoReply )) then 0
           else -1 }"/>
         <xs:element name="Checksum" type="pcap:bit" dfdl:length="16" />
-        <xs:choice>
-          <xs:element name="EchoRequest">
+        <xs:choice dfdl:choiceDispatchKey="{
+          if ((./Type eq 8) and (./Code eq 0)) then 'echoRequest'
+          else if ((./Type eq 0) and (./Code eq 0)) then 'echoReply'
+          else fn:error('Not echo request nor echo reply.')
+          }">
```

Comment:
Can we concat Type and Code to create a key and removw the if-else-if?

Reply:
will do